src/engine/renderer/tr_init.cpp

```diff
@@ -661,82 +646,84 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 	R_TakeScreenshot
 	==================
 	*/
-	void R_TakeScreenshot( const char *name, ssFormat_t format )
+	static bool R_TakeScreenshot( Str::StringRef path, ssFormat_t format )
 	{
-		static char         fileName[ MAX_OSPATH ]; // bad things may happen if two screenshots per frame are taken.
-		ScreenshotCommand  *cmd;
-		int                 lastNumber;
-
-		cmd = R_GetRenderCommand<ScreenshotCommand>();
+		ScreenshotCommand *cmd = R_GetRenderCommand<ScreenshotCommand>();
 
 		if ( !cmd )
 		{
-			return;
+			return false;
 		}
 
-		if ( ri.Cmd_Argc() == 2 )
-		{
-			Com_sprintf( fileName, sizeof( fileName ), "screenshots/" PRODUCT_NAME_LOWER "-%s.%s", ri.Cmd_Argv( 1 ), name );
-		}
-		else
-		{
-			qtime_t t;
+		cmd->x = 0;
+		cmd->y = 0;
+		cmd->width = glConfig.vidWidth;
+		cmd->height = glConfig.vidHeight;
+		Q_strncpyz(cmd->fileName, path.c_str(), sizeof(cmd->fileName));
+		cmd->format = format;
 
-			ri.RealTime( &t );
+		return true;
+	}
 
-			// scan for a free filename
-			for ( lastNumber = 0; lastNumber <= 999; lastNumber++ )
+namespace {
+	class ScreenshotCmd : public Cmd::StaticCmd {
```

Comment:
personal preference but we shouldn't indent namespaces.

Reply:
Dedented.